force_wfmanager/wfmanager_task.py

```diff
@@ -54,13 +57,25 @@ def create_dock_panes(self):
 
     def save_workflow(self):
         """ Shows a dialog to save the workflow into a JSON file """
-        dialog = FileDialog(action="save as", default_filename="workflow.json")
-        result = dialog.open()
+        writer = WorkflowWriter()
 
-        if result is OK:
-            writer = WorkflowWriter()
-            with open(dialog.path, 'wr') as output:
-                writer.write(self.workflow_m, output)
+        # If the user already saved before or loaded a file, we overwrite this
+        # file
+        if len(self.current_file) != 0:
+            current_file = self.current_file
+        else:
+            dialog = FileDialog(
+                action="save as", default_filename="workflow.json")
+            result = dialog.open()
+
+            if result is not OK:
+                return
+
+            current_file = dialog.path
+
+        with open(current_file, 'w') as output:
```

Comment:
You need to handle exception here. Otherwise the first time the file can't be opened you get no response at all, the user thinks everything has been saved, while in reality there's a big exception in the console he can't see.